polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -165,7 +158,7 @@ class MemorySelector {
   DenseMap<Value, SmallVector<affine::MemRefAccess>> accesses;
 
   /// The preferred memory space for each memref access;
-  DenseMap<const affine::MemRefAccess *, MemorySpace> accessToMemSpace;
+  DenseMap<const Operation *, MemorySpace> accessToMemSpace;
```

Comment:
address of MemRefAccess are different when insert and when query, as they are created locally, need to use the Operation `access.opInst` instead.